provisioning_templates/snippet/kickstart_networking_setup.erb

```diff
@@ -8,35 +8,43 @@ description: this will configure your host networking, it configures your primar
 %>
 <% subnet = @host.subnet -%>
 <% subnet6 = @host.subnet6 -%>
-<% dhcp = subnet.dhcp_boot_mode? -%>
+
+rm -v -f /etc/sysconfig/network-scripts/ifcfg-eth*
+rm -v -f /etc/sysconfig/network-scripts/ifcfg-eno*
+rm -v -f /etc/sysconfig/network-scripts/ifcfg-ens*
+rm -v -f /etc/udev/rules.d/70-persistent-net.rules
 
 <% if @host.respond_to?(:has_primary_interface?) %>
 <%# Foreman 1.7 - primary interface contained in @host %>
 # primary interface
-real=`ip -o link | grep <%= @host.mac -%> | awk '{print $2;}' | sed s/://`
 <% if @host.has_primary_interface? %>
+real=<%= @host.primary_interface.identifier %>
+
+echo "Writing /etc/sysconfig/network-scripts/ifcfg-$real"
 cat << EOF > /etc/sysconfig/network-scripts/ifcfg-$real
-BOOTPROTO="<%= dhcp ? 'dhcp' : 'none' -%>"
```

Comment:
An issue I have with the network management, and I don't know yet how to correctly fix it:
Our mgmt network (usually eth0) is ipv6 only, but for practical reasons (one of those being that pxeboot/kickstart on ipv6 is still a not working for us) we do hand out ipv4 addresses via dhcp.

Basically: at boot time, a server gets an ipv4 address, during kickstart we set a fixed ipv6 address (and _no_ ipv4 address)

I could not model this behaviour in the then current options of Foreman (1.13.2), has this changed? Easiest "fix" was to put the v4 network for this subnet to dhcp, but always set "none" here.